templates/repo/pulls/status.tmpl

```diff
@@ -23,14 +27,22 @@
 			{{ctx.Locale.Tr "repo.pulls.status_checking"}}
 		{{end}}
 
-		{{if .ShowHideChecks}}
 		<div class="ui right">
-			<button class="commit-status-hide-checks btn interact-fg"
-			data-show-all="{{ctx.Locale.Tr "repo.pulls.status_checks_show_all"}}"
-			data-hide-all="{{ctx.Locale.Tr "repo.pulls.status_checks_hide_all"}}">
-			{{ctx.Locale.Tr "repo.pulls.status_checks_hide_all"}}</button>
+			{{if and .StatusCheckData .StatusCheckData.RequireApproval .StatusCheckData.CanApprove}}
+				<form method="post" action="{{.StatusCheckData.ApproveActionLink}}">
```

Comment:
You can use `form-fetch-action`, then no need to do `redirect` trick, the page will be reload automatically, and  the users can see loading indicator when they clicks the button.